packages/forklift-console-plugin/src/plans/create/CreatePlanWizardFooter.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, MouseEvent } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useHistory } from 'react-router';
+import { getResourceUrl } from 'src/modules';
+
+import { PlanModelRef } from '@kubev2v/types';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  useWizardContext,
+  WizardFooterProps,
+  WizardFooterWrapper,
+} from '@patternfly/react-core';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import { PlanWizardStepId } from './constants';
+
+type CreatePlanWizardFooterProps = Partial<Pick<WizardFooterProps, 'nextButtonText' | 'onNext'>>;
+
+export const CreatePlanWizardFooter: FC<CreatePlanWizardFooterProps> = ({
+  nextButtonText,
+  onNext,
+}) => {
+  const history = useHistory();
+  const { t } = useForkliftTranslation();
+  const [activeNamespace] = useActiveNamespace();
+  const { trigger } = useFormContext();
+  const { activeStep, goToNextStep, goToPrevStep, goToStepById } = useWizardContext();
+  const canSkipToReview =
+    activeStep.id === PlanWizardStepId.MigrationType ||
+    activeStep.id === PlanWizardStepId.OtherSettings;
+
+  const validate = async () => trigger(null, { shouldFocus: true });
+
+  const onNextClick = async (event: MouseEvent<HTMLButtonElement>) => {
+    if (onNext) {
+      return onNext(event);
+    }
+
+    const isValid = await validate();
+    if (isValid) {
+      goToNextStep();
+    }
+  };
+
+  const onSkipToReviewClick = async (event: MouseEvent<HTMLButtonElement>) => {
+    if (onNext) {
+      return onNext(event);
+    }
```

Comment:
this can be inside valid also? 
 valid func will have a different name (as its not only valid anymore), but can return the value of trigger()